### DISCUSSION TITLE
Trying to set up WalletConnectConnector

### DISCUSSION THREAD

**Discussion by evankalaw** [wevm/wagmi#472] [Q&A] [answered]
Hey there,

I am currently trying to add support to my application for WalletConnect -> MetaMask (using an Injected connector) is working just as expected and we've run into no issues.

However, I keep getting this error when adding a WalletConnectConnector to my WagmiClient:
<img width="376" alt="Screen Shot 2022-05-17 at 5 11 25 PM" src="https://user-images.githubusercontent.com/13758377/168919544-dbef2e96-45b7-4616-8b37-35cca226d5b4.png">

This error is popping up when hitting my 'connect wallet' button, which is calling connectAsync() on the appropriate connector, which is the same functionality for when just MetaMask was supported.

When initializing my WalletConnectConnector inside my WagmiClient, it looks like this:

<img width="671" alt="Screen Shot 2022-05-17 at 5 12 56 PM" src="https://user-images.githubusercontent.com/13758377/168919960-c3577c57-4506-496a-bcce-c5ed8975d528.png">

appChain, etc, are set up because currently this is being tested on a singular ETH test network. Am I missing something here?

**Comment by tmm** [comment, marked as answer]
Depending on your framework, you might need to provide polyfills for `buffer` (and other Node internals). See https://github.com/tmm/wagmi/discussions/163 for more info.

**Reply by evankalaw** [reply]
Dope! I am using React w/ Typescript and followed those instructions and it looks good to me bro. Thanks:
<img width="496" alt="Screen Shot 2022-05-17 at 6 34 23 PM" src="https://user-images.githubusercontent.com/13758377/168928839-51f4d99f-6ccc-4d77-b8b2-8a4ed3fe2e3d.png">
